mevislab.github.io/content/tutorials/thirdparty/matplotlib/modulesetup.md

```diff
@@ -0,0 +1,89 @@
+
+Your finished network should look like this: 
+
+![Base network](/images/tutorials/thirdparty/Matplotlib1.PNG)
+
+{{< networkfile "/tutorials/thirdparty/matplotlib/MatplotlibBaseNetwork.mlab" >}}
+
+### Excursion on the concept behind modules
+
+To be able to build on the foundation we just set, it can be useful to understand how modules are conceptualized:
+You will have noticed how, for every module, a panel will pop up if you double-click it. The modules panel contains all of its functional parameters and enables you, as the user, to change them within a graphical user interface (GUI). We will do something similar later on. 
+But where and how is a module panel created? To answer this question, please close the module panel and right-click on the module. 
+A context menu will open, click on "Related Files". 
+
+![Context menu of the "SubImage" module](/images/tutorials/thirdparty/Matplotlib2.png)
+
+As you can see, each module has a .script and a .py file, named like the module itself: 
++ The .script file is, where the appearance and structure of the module panel as well as their commands are declared. 
++ The .py file contains Python functions and methods, whose are triggered by their referenced commands within the .script file.
+
+Some modules also reference a .mlab file which usually contains their internal network as the module is a macro. 
+
+**Let's continue with our module setup now:**  
+If your network is ready, group it by right-clicking on your group's title and select "Grouping", then "Add To A New Group".
+After, convert your grouped network into a macro module. 
+
+![Converting to a macro](/images/tutorials/thirdparty/Matplotlib3.png)
+{{<alert class="info" caption="Info">}}
+Information on how to convert groups into macros can be found [here](/tutorials/basicmechanisms#TutorialMacroModules). 
+{{</alert>}}
+Depending on whether or whether not you like to reuse your projects in other workspaces, it can make sense to convert to a global macro.
+We'd recommend to do so. 
+
+Now open the .script file of your newly created macro through the context menu. The file will be opened within MATE (MeVisLab Advanced Text Editor). Add this short piece of code into your .script file and make sure that the .script and the .py are named exactly the same as the module they are created for.
+
+    Commands{
+      source = $(LOCAL)/BaseNetwork.py
+    }
+
+
+Click the "Reload" button that is located above the script for the .py file to be added into the module definition folder, then open it using the "Files" button on the same bar as demonstrated below:
+![MATE](/images/tutorials/thirdparty/Matplotlib5.png)
+
+{{<alert class="info" caption="Info">}}
+The [MDL Reference](https://mevislabdownloads.mevis.de/docs/current/MeVisLab/Resources/Documentation/Publish/SDK/MDLReference/index.html) is a very handy tool for this and surely also for following projects. 
```

Comment:
surely?

Reply:
"used to emphasize the speaker's firm belief that what they are saying is true and often their surprise that there is any doubt of this:
"if there is no will, then surely the house goes automatically to you""
I think that they'll want to reuse the reference in the future for other projects.